index.js

```diff
@@ -436,6 +436,68 @@ const waitOnMaybe = () => {
 
 const I = (x) => x
 
+const detectPrNumber = async () => {
+  const {
+    GITHUB_SHA,
+    GITHUB_TOKEN,
+    GITHUB_RUN_ID,
+    GITHUB_REPOSITORY,
+    GITHUB_HEAD_REF,
+    GITHUB_REF,
+    GITHUB_SERVER_URL,
+    CYPRESS_PULL_REQUEST_ID,
+    CYPRESS_PULL_REQUEST_URL
+  } = process.env
+
+  if (CYPRESS_PULL_REQUEST_ID && CYPRESS_PULL_REQUEST_URL) {
+    // Both pull request envs are already defined - no need to do anything else
+    return
+  }
+
+  const [owner, repo] = GITHUB_REPOSITORY.split('/')
+  let prNumber
+
+  if (GITHUB_TOKEN) {
+    debug(
+      `Detecting PR number by asking GitHub about run ${GITHUB_RUN_ID}`
+    )
+
+    const client = new Octokit({
+      auth: GITHUB_TOKEN
+    })
+
+    if (GITHUB_HEAD_REF) {
+      // GITHUB_HEAD_REF is only defined when the event that triggered it was 'pull_request' or 'pull_request_target' (meaning a PR number should be readily-available)
+      // should have format refs/pull/<pr_number>/merge when triggered by pull_request workflow
+      prNumber = parseInt(GITHUB_REF.split('/')[2])
+    } else {
+      const resp = await client.request(
```

Comment:
Might also be worth adding a try/catch around this request in the event of a partial outage or something. I don't think we'd want to halt a recording in that case.